frontend/src/slices/auth/libs/hooks/use-auth-user.hook.ts

```diff
@@ -1,9 +1,18 @@
 import { useAppSelector } from '~/libs/hooks/hooks.js';
+import { type UserEntityObjectWithGroupT } from '~/libs/types/types.js';
 
 import { selectUser } from '../../selectors.js';
 
-const useAuthUser: () => ReturnType<typeof selectUser> = () => {
-  return useAppSelector(selectUser);
+const useAuthUser = <
+  T extends ReturnType<typeof selectUser> = UserEntityObjectWithGroupT | null,
+>(): T => {
+  const user = useAppSelector(selectUser);
+
+  if (!user) {
+    return null as T;
```

Comment:
No, i want that you can use it like so:
const user = useAuthUser<UserEntityObjectWithGroupT>();

the type of the user there will be UserEntityObjectWithGroupT, not UserEntityObjectWithGroupT | null, because i want to use it in protected routes and don't want to have redundant if(user) {then ...} check all over the place.

In fact, because I haven't done it like this before, we actually have this situation with if(user) {then ...} everywhere.